packages/ERTP/src/paymentLedger.js

```diff
@@ -178,6 +186,9 @@ export const preparePaymentLedger = (
    * @param {SetStore<Payment>} [optRecoverySet]
    */
   const initPayment = (payment, amount, optRecoverySet = undefined) => {
+    if (recoverySetsState === 'noRecoverySets') {
+      assert(optRecoverySet === undefined);
+    }
     if (optRecoverySet !== undefined) {
```

Comment:
@Chris-Hibbert , as a first step, what if we changed this line to
```suggestion
    if (optRecoverySet !== undefined && !AmountMath.isEmpty(amount)) {
```
?

This would at least stop new zero payments from being added to recovery sets.

Reply:
This seems worthwhile to me.

Reply:
Done